### ISSUE TITLE
Feedback on naming patterns and how to encourage best practices

### ISSUE THREAD

**Issue by annezazu** [WordPress/gutenberg#40033]
## What problem does this address?

This came up as part of the [twelfth call for testing for the FSE Outreach Program:](https://make.wordpress.org/test/2022/02/23/fse-program-testing-call-12-hyping-headers/#comment-2377)

>This is nitpicky, but I think there should be more care into the naming of Header Patterns (or any pattern). Since I had changed the default color scheme on my site, I was confused at first looking at the pattern names. For example, I had set my primary color as Blue, so “Text-only header with green background” was confusing to me at first because it was blue. Or “Text-only header with salmon background” was the gray I had chosen for my site. I think it’s cool that it respects your site color pallet, it was just confusing to me for a second. Took a while for my brain to catch back up.

For more context, the key here is that the default colors for a theme can now be overridden thanks to Styles. As a result, the naming of patterns like “Text-only header with green background” don't make sense if you've updated the green to a blue. Here are some visuals to clarity:

**Default Colors**
![Screen Shot 2022-04-04 at 10 44 58 AM](https://user-images.githubusercontent.com/26996883/161609416-8b6c4296-9e80-4611-baa5-56c3d12d0973.png)

**After Updating Default Colors**
![Screen Shot 2022-04-04 at 11 09 57 AM](https://user-images.githubusercontent.com/26996883/161609421-0573845a-d749-4c88-bb5d-c62aa6ee1a80.png)

**Patterns After Updating Colors**
![Screen Shot 2022-04-04 at 11 10 27 AM](https://user-images.githubusercontent.com/26996883/161609423-f5da2a50-447b-479b-86f9-2b3d6e5f8457.png)

## What is your proposed solution?

In an ideal world, we can encourage folks to name patterns based on semantic colors. For the example of above, it would be something like “Text-only header with secondary background”. Curious what others think though as this is a bit of a tricky/odd problem :) cc'ing a few folks below. 

@jffng for considering updating TT2 pattern names specifically. 
@bph for dev docs to consider encouraging folks creating patterns to think about this use case when it comes to naming.
@carolinan for general thoughts since you're always so thoughtful with these sorts of problems.

**Comment by jffng** [comment]
Thanks for the ping. It is a tricky problem, and the naming is not ideal when a user changes the palette.

I'm not quite sure how this should be addressed. Take for example the pattern _Header with centered logo and black background_. If we switched to semantic naming mapping to the way the color presets are named, this would become _Header with centered logo and foreground background_, which is maybe more confusing 😅   

Should we go with something more generic, like appending an "alt" or " — A"? cc @kjellr

**Comment by kjellr** [comment]
Yeah, this is a tough problem. 😩 I think after a while, these labels are relatively unhelpful anyway. I think we should switch to just "with background color" and avoid doing the A, B, C  thing.

**Comment by carolinan** [comment]
Are these patterns in the core (would need a trac issue) or the theme (would also need a trac issue)?

**Comment by carolinan** [comment]
I have created a trac ticket for Twenty Twenty-Two here: [ 56720 ](https://core.trac.wordpress.org/ticket/56720).
